### PR TITLE
Import docx, convert to markdown, split into files

### PR DESCRIPTION
**Pull request by arthurattwell**
**For team review, not yet for merging.**

This adds the ability to import a .docx file, converting it to markdown, and splitting it into separate markdown files with basic YAML frontmatter. Documented in `_docs/setup/quick-start.md` currently, though perhaps importing should have its own separate doc.

### CODE REVIEW

File: _docs/setup/quick-start.md [hunk 1]
```diff
@@ -19,10 +19,39 @@ This quick setup assumes you already have [Jekyll](https://jekyllrb.com/), [Node
...
+   - If you're adding to an existing book, run `npm run eb -- text --book existingbookname --source myworddoc.docx`
+
+   In those examples, replace `myworddoc` with your .docx file's name, and `existingbookname` with the name of your existing book folder.
+3. If you want to split a markdown file into separate files, and that file is already in its book folder, run `npm run eb -- text --book existingbookname --source markdownfile.md --split`.
+
+   Alternatively, you can add the `--split` argument to the `npm run eb -- new --source myworddoc.docx` command to split when importing it.
+
+### The split marker
+
+By default, the file will be split at each first-level heading marked with `#` in the document. You can specify a different string to split on, like `##`, by specifying it in quotes after `--split`, e.g. `--split '##'`.
```
[the remark this answers is not in the document]
Comment (arthurattwell) [reply _docs/setup/quick-start.md]: Thanks, great point. I've now expanded the docs on importing and splitting, and have covered this there.

File: _tools/run/commands/text.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+exports.desc = 'Process text'
+exports.handler = function (argv) {
+  'use strict'
+
+  // What file type are we processing?
+  const filetype = argv.source.split('.').pop()
+
+  if (filetype) {
+    if (filetype === 'docx') {
+      // If the source is a Word file, run importText
```
[the remark this answers is not in the document]
Comment (arthurattwell) [reply _tools/run/commands/text.js]: Thanks, nice catch. Fixed.

File: _tools/run/helpers/helpers.js [hunk 3]
```diff
@@ -1663,6 +1664,271 @@ function newBook (argv) {
...
+      }
+
+      pandoc(sourceFile, pandocArgs, pandocCallback)
+    }
+  } catch (error) {
+    console.error('Unable to convert ' + argv.source)
+  }
+}
+
+// Generate a copy-pasteable file list in a file
```
[the remark this answers is not in the document]
Comment (arthurattwell) [reply _tools/run/helpers/helpers.js:1759]: Good idea, I've added that in the expanded docs I've now added.

File: _tools/run/helpers/helpers.js [hunk 4]
```diff
@@ -1663,6 +1664,271 @@ function newBook (argv) {
...
+
+    // Create a counter for every filePart,
+    // and a counter for those we actually use.
+    let filePartCounter = 0
+    let bookPartCounter = 0
+
+    // Write each filepart to a new file
+    filePartsArray.forEach(function (filePart) {
+
+      // Create a filename from the first line
```
[the remark this answers is not in the document]
Comment (arthurattwell) [reply _tools/run/helpers/helpers.js:1847]: Good idea, done.

File: _tools/run/helpers/helpers.js [hunk 5]
```diff
@@ -1663,6 +1664,271 @@ function newBook (argv) {
...
+
+        // If the file has no first line for a slug,
+        // do not use a separator for the filename
+        let fileNameSeparator = ''
+        if (firstLine && firstLineSlug) {
+          fileNameSeparator = '-'
+        }
+
+        // Write the file
+        const newFileName = newFileNumber + fileNameSeparator + firstLineSlug + '.md'
```
[the remark this answers is not in the document]
Comment (arthurattwell) [reply _tools/run/helpers/helpers.js:1904]: I think there may be cases where leaving the digits is desired behaviour. E.g. if a novel has a prologue, that might be `01-prologue` and the first chapter is then `02-1-the-first-chapter`. So the numbering we're adding is only for ordering purposes, and other numbers may be for reference or readability. So I'm leaving this unchanged for now.

**Review by LouiseSteward — APPROVED** [review]

**Pull request merged**